python/ray/autoscaler/aws/node_provider.py

```diff
@@ -126,6 +126,25 @@ def create_node(self, node_config, tags, count):
                 "Key": k,
                 "Value": v,
             })
+        tag_specs = [{
+            "ResourceType": "instance",
+            "Tags": tag_pairs,
+        }]
+        user_tag_specs = conf.get("TagSpecifications", [])
+        for user_tag_spec in user_tag_specs:
```

Comment:
add a comment: Allow users to override tags with their own

Reply:
Hm does this actually allow "Name" to be overriden?

Reply:
We iterate over all the tag specs provided by the user. We only override the 'instance' tag spec. All other tag specs are appended, because `instance` is the only resource type Ray provides by default. With that said, the only thing we need to check is that the tags for the user-specified `instance` resource type are being overridden properly.

Below, we iterate over the user-specified tags if resource type is `instance`. We check if the tag exists. If it does, we only set the value provided by the user for that tag (e.g. `Name`). If it doesn't exist, we append the tag to the list of `instance` tags.